coding_systems/base/import_data_utils.py

```diff
@@ -136,3 +149,118 @@ def setup_new_import_database(self):
+
+    Note we don't exlude version created from or already in the compatible
+    set for THIS release.  If we're doing a forced-overwrite import (which is where
+    we'd expect that might happen), we want to re-check compatibility
+
+    Then check
+    1) versions with searches by rerunning the searches with the new release and checking
+       the results are the same
+    2) versions without searches, but with hierarchies, by recreating the hierarchy with the
+       new release and checking if it is identical to the version's cached hierarchy
+    """
+    previous_release = (
+        CodingSystemRelease.objects.filter(coding_system=coding_system_id)
+        .exclude(id=new_coding_system.release.id)
+        .latest("valid_from")
+    )
+    # Filter to versions compatible with previous release
+    versions_to_check = CodelistVersion.objects.filter(
+        Q(compatible_releases__id=previous_release.id)
+        | Q(coding_system_release=previous_release)
+    )
+    print(
+        f"Found {len(versions_to_check)} versions compatible with previous release '{previous_release.database_alias}'."
+    )
+    # Find any versions with searches
+    print("Checking compatibility by search:")
+    versions_with_searches = set(
+        versions_to_check.filter(
+            codelist__coding_system_id=coding_system_id, searches__isnull=False
+        )
+    )
+    compatible_count, total = check_and_update_compatibile_versions(
+        new_coding_system, versions_with_searches, _check_version_by_search
+    )
+    print(f"{total} checked; {compatible_count} identified as compatible")
+
+    # check any versions without searches, but with hierarchies
+    print("Checking compatibility by hierarchy:")
+    versions_with_hierarchy = {
+        version
+        for version in versions_to_check.filter(
+            codelist__coding_system_id=coding_system_id, searches__isnull=True
+        )
+        if version.has_hierarchy
+    }
+    compatible_count, total = check_and_update_compatibile_versions(
+        new_coding_system, versions_with_hierarchy, _check_version_by_hierarchy
+    )
+    print(f"{total} checked; {compatible_count} identified as compatible")
+
+
+def check_and_update_compatibile_versions(
+    coding_system, versions, compatibility_check_function
+):
+    """Check compatibility of codelist versions against a specific coding system release"""
+    compatible_count = 0
+    number_of_versions = len(versions)
+    for counter, version in enumerate(versions, start=1):
+        update_progress(counter, number_of_versions, comment="Checking: ")
+        is_compatible = compatibility_check_function(coding_system, version)
+        if is_compatible:
+            version.compatible_releases.add(coding_system.release)
+            compatible_count += 1
+    return compatible_count, number_of_versions
+
+
+def _check_version_by_search(coding_system, version):
```

Comment:
This needs a docstring explaining why the searches being the same implies compatibility.  (Although actually I think it doesn't -- see above.)